### PR TITLE
Subfeature/preprocessor test

### PR DESCRIPTION
**Pull request by Ragingram2**
Mostly autogenerated make_prototype and make_reflector functions.
But also some features that support custom attribute serialization for member variables and structs.
https://github.com/Rythe-Interactive/RythePreprocessor/tree/CPPAstNet
Located above is the repo for the preprocessor. Before launching, open the project and add the directory to your rythe repo to the command line arguments along with some flags for the indivudal modules, and applications you want to include in the autogen.

### CODE REVIEW

File: applications/sandbox/autogen/autogen_prototype_example_comp.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include "autogen_prototype_example_comp.hpp"
```
Comment (LeonBrands) [inline applications/sandbox/autogen/autogen_prototype_example_comp.cpp]: Bit of a general question but shouldn't autopen files be gitignored? Seems a bit excessive and might cause conflicts more easily

File: legion/engine/core/core.hpp [hunk 2]
```diff
@@ -29,6 +29,7 @@
 #include <core/engine/module.inl>
 #include <core/serialization/serialization.hpp>
 
+
```
Comment (LeonBrands) [inline legion/engine/core/core.hpp]: ?

File: legion/engine/rendering/data/model.cpp [hunk 3]
```diff
@@ -233,7 +233,7 @@ namespace legion::rendering
                     material.set_param("useHeight", false);
                 }
 
-                material.setLoadOrSaveBit(false);
```
Comment (LeonBrands) [inline legion/engine/rendering/data/model.cpp:236]: For all changes in this file: Remove commented out code

**Review by LeonBrands — CHANGES_REQUESTED** [review]

File: applications/sandbox/autogen/autogen_prototype_example_comp.cpp (hunk 1 above)
Comment (Ragingram2) [reply applications/sandbox/autogen/autogen_prototype_example_comp.cpp]: yeah ur right, they should be

**Pull request closed without merging**